lib/puppet/pops/functions/function.rb

```diff
@@ -47,18 +47,46 @@ def call(scope, *args)
   # Allows the implementation of a function to call other functions by name. The callable functions
   # are those visible to the same loader that loaded this function (the calling function).
   #
```

Comment:
It is worth mentioning that this calls the referenced function with the calling functions closure scope as the caller's scope (typically "global scope").